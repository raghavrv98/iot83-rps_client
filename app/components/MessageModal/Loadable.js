/**
 *
 * Asynchronously loads the component for MessageModal
 *
 */

import loadable from "@loadable/component";

export default loadable(() => import("./index"));
