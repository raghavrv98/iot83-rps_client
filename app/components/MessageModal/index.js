/**
 *
 * MessageModal
 *
 */

import React from "react";
import { FormattedMessage } from "react-intl";
import commonMessages from "../../messages";
// import PropTypes from 'prop-types';
// import styled from 'styled-components';

function MessageModal(props) {
	return <div className="modal fade show d-block" id="modal">
		<div className="modal-dialog">
			<div className="modal-content">
				{props.type === "confirm" ?
					<div className="modal-body p-4 pd-t-30">
						<span className="close" onClick={() => { props.onClose() }}><i className="far fa-times"></i></span>
						<h5 className="text-center mb-5">{props.message}</h5>
						<div className="text-right">
							<button type="button" className="btn btn-danger" data-dismiss="modal" onClick={() => { props.onConfirm() }}><FormattedMessage {...commonMessages.confirmButton} /></button>
							<button type="button" className="btn btn-secondary mr-l-10" data-dismiss="modal" onClick={() => { props.onClose() }}><FormattedMessage {...commonMessages.cancelButton} /></button>
						</div>
					</div> :
					<div className="modal-body p-0 text-center position-relative">
						{props.type === "success" ?
							<div className="alert alert-success mb-0 p-5">
								<span className="close" onClick={() => { props.onClose() }}><i className="far fa-times"></i></span>
								<p className="text-success m-0 f-16">{props.message}</p>
							</div>
							:
							<div className="alert alert-danger mb-0 p-5">
								<span className="close" onClick={() => { props.onClose() }} ><i className="far fa-times"></i></span>
								<p className="text-danger m-0 f-16">{props.message}</p>
							</div>
						}
					</div>
				}
			</div>
		</div>
	</div>;
}

MessageModal.propTypes = {};

export default MessageModal;
