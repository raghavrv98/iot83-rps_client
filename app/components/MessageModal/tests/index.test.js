import React from 'react';
import { shallow } from 'enzyme';
// import { enzymeFind } from 'styled-components/test-utils';

import MessageModal from '../index';

describe('< MessageModal />', () => {

    it('match snapshots', () => {
        const wrap = shallow(<MessageModal type="confirm" />)
        expect(wrap).toMatchSnapshot()
    })
    
    it("test children count for success or error" , () => {
        const wrap = shallow(<MessageModal type="suceess" />)
        let modalBody = wrap.find("div.modal-body");
        expect(modalBody.children().length).toBe(1);
    })

    it("test children count for confirm or error" , () => {
        const wrap = shallow(<MessageModal type="confirm" />)
        let modalBody = wrap.find("div.modal-body");
        expect(modalBody.children().length).toBe(3);
    })

    it("test function in case of confirm modal" , () => {
        const closeFunction = jest.fn();
        const confirmFunction = jest.fn();
        const wrap = shallow(<MessageModal type="confirm" onClose={closeFunction} onConfirm={confirmFunction} />)
        let closeSpan = wrap.find("span.close");
        closeSpan.simulate('click')
        expect(closeFunction.mock.calls.length).toBe(1);
        let closeButton = wrap.find("button.btn-secondary");
        closeButton.simulate('click')
        expect(closeFunction.mock.calls.length).toBe(2);
        let confirmButton = wrap.find("button.btn-danger");
        confirmButton.simulate('click')
        expect(confirmFunction.mock.calls.length).toBe(1);
    })
    it("test function in case of success modal" , () => {
        const closeFunction = jest.fn();
        const wrap = shallow(<MessageModal type="success" onClose={closeFunction} />)
        let closeSpan = wrap.find("span.close");
        closeSpan.simulate('click')
        expect(closeFunction.mock.calls.length).toBe(1);
    })
    it("test function in case of error modal" , () => {
        const closeFunction = jest.fn();
        const wrap = shallow(<MessageModal type="error" onClose={closeFunction} />)
        let closeSpan = wrap.find("span.close");
        closeSpan.simulate('click')
        expect(closeFunction.mock.calls.length).toBe(1);
    })
});