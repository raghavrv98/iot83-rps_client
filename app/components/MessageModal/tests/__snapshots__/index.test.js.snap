// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`< MessageModal /> match snapshots 1`] = `
<div
  className="modal fade show d-block"
  id="modal"
>
  <div
    className="modal-dialog"
  >
    <div
      className="modal-content"
    >
      <div
        className="modal-body p-4 pd-t-30"
      >
        <span
          className="close"
          onClick={[Function]}
        >
          <i
            className="far fa-times"
          />
        </span>
        <h5
          className="text-center mb-5"
        />
        <div
          className="text-right"
        >
          <button
            className="btn btn-danger"
            data-dismiss="modal"
            onClick={[Function]}
            type="button"
          >
            <FormattedMessage
              defaultMessage="Confirm"
              id="app.common.messages.confirmButton"
              values={Object {}}
            />
          </button>
          <button
            className="btn btn-secondary mr-l-10"
            data-dismiss="modal"
            onClick={[Function]}
            type="button"
          >
            <FormattedMessage
              defaultMessage="Cancel"
              id="app.common.messages.cancelButton"
              values={Object {}}
            />
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
